config/config.go

```diff
@@ -40,6 +40,10 @@ const (
 	MaxLogFileSize = 4096 // MB
 	// DefTxnTotalSizeLimit is the default value of TxnTxnTotalSizeLimit.
 	DefTxnTotalSizeLimit = 100 * 1024 * 1024
+	// DefMaxIndexLength is the maximum index length. This value is consistent with MySQL.
+	DefMaxIndexLength = 3072
+	// DefTiDBMaxIndexLength is the maximum index length for TiDB v3.0.7 and previous version.
+	DefTiDBMaxIndexLength = 3072 * 4
```

Comment:
Name it after the convention like 'maxAllowedPacket' and 'maxOfMaxAllowedPacket'.